dss/__init__.py

```diff
@@ -34,6 +38,26 @@ def get_logger():
     except RuntimeError:
         return logging.getLogger(__name__)
 
+class OperationWithAuthorizer(Operation):
+    authorized_domains = os.environ["AUTHORIZED_DOMAINS"].split()
+    def oauth2_authorize(self, function):
+        def wrapper(request):
+            if "token_info" in request.context.values:
+                token_info = request.context.values["token_info"]
+                if not int(token_info["expires_in"]) > 0:
+                    raise OAuthProblem(description="Authorization token has expired")
```

Comment:
After rebasing, I found that this does indeed interact with `common_error_handler`. So I special-cased OAuth errors in `common_error_handler`. I don't think the spec is consulted.